sdk/servicebus/azure-servicebus/azure/servicebus/management/_management_client.py

```diff
@@ -0,0 +1,287 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from contextlib import contextmanager
+from copy import copy
+from typing import TYPE_CHECKING, Dict, Any, Union, List, cast, Type
+from xml.etree.ElementTree import ElementTree, Element
+
+from msrest.serialization import Model
+from msrest.exceptions import ValidationError
+from azure.core.exceptions import ResourceNotFoundError, HttpResponseError, raise_with_traceback
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.policies import HttpLoggingPolicy, DistributedTracingPolicy, ContentDecodePolicy, \
+    RequestIdPolicy, BearerTokenCredentialPolicy
+from azure.core.pipeline.transport import RequestsTransport
+
+from .._common.constants import JWT_TOKEN_SCOPE
+from .._common.utils import parse_conn_str
+from .._base_handler import ServiceBusSharedKeyCredential
+from ._shared_key_policy import ServiceBusSharedKeyCredentialPolicy
+from ._generated._configuration import ServiceBusManagementClientConfiguration
+from ._generated.models import CreateQueueBody, CreateQueueBodyContent, \
+    QueueDescription, QueueRuntimeInfo, MessageCountDetails
+from ._generated._service_bus_management_client import ServiceBusManagementClient as ServiceBusManagementClientImpl
+from ._model_workaround import QUEUE_DESCRIPTION_SERIALIZE_ATTRIBUTES, avoid_timedelta_overflow
+from . import _constants as constants
+
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential  # pylint:disable=ungrouped-imports
+
+
+@contextmanager
+def _handle_response_error():
+    try:
+        yield
+    except HttpResponseError as response_error:
+        try:
+            new_response_error = HttpResponseError(
+                message=response_error.model.detail,
+                response=response_error.response,
+                model=response_error.model
+            )
+        except AttributeError:
+            new_response_error = response_error
+        raise new_response_error
+
+
+def _convert_xml_to_object(queue_name, et, class_):
+    # type: (str, Union[Element, ElementTree], Type[Model]) -> Union[QueueDescription, QueueRuntimeInfo]
+    content_ele = cast(ElementTree, et).find(constants.CONTENT_TAG)
+    if not content_ele:
+        raise ResourceNotFoundError("Queue '{}' does not exist".format(queue_name))
```

Comment:
I know we've only done queue-specific work thus far, but this seems like a generalized function from the name, but has queue-specific verbiage internally.  My naive response would be that one or the other should change.  (either this should be an explicitly queue-specific conversion function, or we should generalize the internals)

Reply:
Yes this might be used for Topic and other entities. Let's change when adding them in next release.

Reply:
I now changed this function to only work on the internal QueueDescription model, which combines the previous QueueDescription and QueueRuntime. 
Then the convenience layer creates its own QueueDescription (not that internal QueueDescription class) and QueueRuntimeInfo objects.